modules/core/src/main/java/org/apache/ignite/internal/processors/compress/CompressionProcessor.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.compress;
+
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.configuration.DiskPageCompression;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.IgniteComponentType;
+import org.apache.ignite.internal.processors.GridProcessorAdapter;
+import org.apache.ignite.internal.processors.cache.persistence.tree.io.PageIO;
+
+/**
+ * Compression processor.
+ *
+ * @see IgniteComponentType#COMPRESSION
+ */
+public class CompressionProcessor extends GridProcessorAdapter {
+    /** */
+    public static final int LZ4_MIN_LEVEL = 0;
+
+    /** */
+    public static final int LZ4_MAX_LEVEL = 17;
+
+    /** */
+    public static final int LZ4_DEFAULT_LEVEL = 0;
+
+    /** */
+    public static final int ZSTD_MIN_LEVEL = -131072;
+
+    /** */
+    public static final int ZSTD_MAX_LEVEL = 22;
+
+    /** */
+    public static final int ZSTD_DEFAULT_LEVEL = 3;
```

Comment:
My recommendation is to use 2 since performance is much better with roughly same compression ratio:
http://apache-ignite-developers.2346864.n4.nabble.com/Compression-prototype-tp34199p34649.html